tests/ui/components/filters/filteroptionscomponent.js

```diff
@@ -233,3 +229,267 @@ describe('filter options component', () => {
     expect(setStaticFilterNodes.mock.calls).toHaveLength(6);
   });
 });
+
+describe('filter options component selected options', () => {
+  let COMPONENT_MANAGER, defaultConfig;
+
+  beforeEach(() => {
+    const bodyEl = DOM.query('body');
+    DOM.empty(bodyEl);
+    DOM.append(bodyEl, DOM.createEl('div', { id: 'test-component' }));
+
+    COMPONENT_MANAGER = mockManager(
+      {
+        globalStorage: {
+          getState: key => {
+            if (key === 'test-previous-options') {
+              return ['label1', 'label2'];
+            }
+          },
+          delete: () => {}
+        }
+      },
+      FilterOptionsComponent.defaultTemplateName()
+    );
+
+    defaultConfig = {
+      container: '#test-component',
+      options: [
+        {
+          label: 'label1',
+          field: 'field',
+          value: 'val1'
+        },
+        {
+          label: 'label2',
+          field: 'field',
+          value: 'val2',
+          selected: false
+        },
+        {
+          label: 'label3',
+          field: 'field',
+          value: 'val3',
+          selected: true
+        },
+        {
+          label: 'label4',
+          field: 'field',
+          value: 'val4',
+          selected: true
+        }
+      ]
+    };
+  });
+
+  it('has no selected options by default', () => {
+    const config = {
+      ...defaultConfig,
+      control: 'singleoption',
+      options: [
+        {
+          label: 'label1',
+          field: 'field',
+          value: 'val1'
+        },
+        {
+          label: 'label2',
+          field: 'field',
+          value: 'val2'
+        },
+        {
+          label: 'label3',
+          field: 'field',
+          value: 'val3'
+        }
+      ]
+    };
+
+    const component = COMPONENT_MANAGER.create('FilterOptions', config);
+    const options = component.config.options;
+    expect(options).toHaveLength(3);
+    expect(options.filter(o => o.selected)).toHaveLength(0);
+  });
+
+  it('can choose selected options for multioption', () => {
+    const config = {
+      ...defaultConfig,
+      control: 'multioption'
+    };
+
+    const component = COMPONENT_MANAGER.create('FilterOptions', config);
+    const options = component.config.options;
+    expect(options).toHaveLength(4);
+    const selectedOptions = options.filter(o => o.selected);
+    expect(selectedOptions).toHaveLength(2);
+    expect(selectedOptions[0].label).toEqual('label3');
+    expect(selectedOptions[1].label).toEqual('label4');
+  });
+
+  it('prioritizes previous options over selected config for multioption', () => {
```

Comment:
'prioritizes previously selected options over selected options in config for multioption'